### PR TITLE
Use LXD with Juju 2.0

### PR DESCRIPTION
**Pull request by huwshimi**
Update the machine view to use LXD containers instead of LXC.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- deploy the GUI with Juju 1.x
- open the machine view and open the form to add a container
- choose LXC and add the container
- you should have a container something like new0/lxc/new1
- deploy the GUI with Juju 2.x
- open the machine view and open the form to add a container
- choose LXD and add the container
- you should have a container something like new0/lxd/new1

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4830/
Test PASSed.

File: jujugui/static/gui/src/app/components/machine-view/add-machine/add-machine.js [hunk 1]
```diff
@@ -138,6 +139,8 @@ YUI.add('machine-view-add-machine', function() {
       @method _generateSelectContainer
     */
     _generateSelectContainer: function() {
+      var lxxOption = this.props.jujuCoreVersion.lastIndexOf('1.', 0) === 0 ?
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/machine-view/add-machine/add-machine.js]: lol, it took me a while reading this as to why it was not a syntax error :)

File: jujugui/static/gui/src/app/components/machine-view/test-machine-view.js [hunk 2]
```diff
@@ -64,6 +64,7 @@ describe('MachineView', function() {
         createMachine={sinon.stub()}
         destroyMachines={sinon.stub()}
         environmentName="My Env"
+        jujuCoreVersion="1.4"
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/machine-view/test-machine-view.js]: We're moving towards setting everything as Juju 2 for the default case. Sorry that's going to make these test changes a little more tedious but this should instead be `jujuCoreVersion="2.0"`. And the rest below.

**Comment by hatched** [comment]
:+1: With the comments addressed.
QA :ok:
Thanks!

**Comment by jaycee** [comment]
With @hatched's comments :+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4836/
Test PASSed.

**Comment by huwshimi** [comment]
Thanks @hatched and @jaycee :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**